test/Microsoft.AspNet.Mvc.FunctionalTests/ConnegTests.cs

```diff
@@ -91,6 +96,58 @@ public class ConnegTests
         }
 
         [Fact]
+        public async Task ProducesAttributeWithTypeOnly_RunsRegularContentNegotiation()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            var expectedContentType = MediaTypeHeaderValue.Parse("application/json;charset=utf-8");
+            var expectedOutput = "{\"Name\":\"John\",\"Address\":\"One Microsoft Way\"}";
+
+            // Act
+            var response = await client.GetAsync("http://localhost/Home/UserInfo_ProducesWithTypeOnly");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(expectedContentType, response.Content.Headers.ContentType);
+            var actual = await response.Content.ReadAsStringAsync();
+            Assert.Equal(expectedOutput, actual);
+        }
+
+        [Fact]
+        public async Task ProducesAttribute_WithTypeAndContentType_UsesContentType()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/xml"));
+            var expectedContentType = MediaTypeHeaderValue.Parse("application/xml;charset=utf-8");
+            var dataContractSerializer = new DataContractSerializer(typeof(User));
+            var stream = new MemoryStream();
+            using (var xmlWriter = XmlWriter.Create(stream, FormattingUtilities.GetDefaultXmlWriterSettings()))
+            {
+                dataContractSerializer.WriteObject(xmlWriter, new User()
+                {
+                    Name = "John",
+                    Address = "One Microsoft Way"
+                });
+            }
```

Comment:
Is the concern related to how we compare xml?...if yes, once this PR (https://github.com/aspnet/Mvc/pull/1962) gets in, it should be simple to compare it...in that case, is it ok for this code to stay as it is?